src/help/help.go

```diff
@@ -64,6 +68,103 @@ func help(topic string) string {
 	if f, present := m[topic]; present {
 		return helpFromBuildRule(f.FuncDef)
 	}
+
+	// Check plugins
+	if pluginHelp := helpForPlugin(topic); pluginHelp != "" {
+		return pluginHelp
+	}
+
+	return ""
+}
+
+func helpForPlugin(topic string) string {
+	config, err := core.ReadDefaultConfigFiles(nil)
+	if err != nil {
+		panic("Failed to read config")
+	}
+	if _, ok := config.Plugin[topic]; ok {
+		message := fmt.Sprintf("${BOLD_BLUE}%v${RESET} is a plugin defined in the .plzconfig file.", topic)
+		if val, ok := config.Plugin[topic].ExtraValues["subrepo"]; ok {
+			subrepo := val[0]
+			buildLabel := core.BuildLabel{PackageName: "", Name: subrepo, Subrepo: subrepo}
+			state := newState()
+
+			// Parse the subrepo (Run reads the plugin config into config)
+			plz.Run([]core.BuildLabel{buildLabel}, nil, state, config, state.TargetArch)
+			downloadBuildLabel := core.BuildLabel{PackageName: "", Name: "_" + subrepo + "#download", Subrepo: ""}
+			if t := state.Graph.Target(downloadBuildLabel); t != nil {
+				if urls := t.AllURLs(state); len(urls) == 1 {
+					message += fmt.Sprintf(" It's loaded from %v\n", urls[0])
+				} else {
+					message += "\n"
+				}
+			} else {
+				message += "\n"
+			}
+
+			state = state.Graph.Subrepo(subrepo).State
+			config := state.Config
+			if config.PluginDefinition.Description != "" {
+				message += "\n" + config.PluginDefinition.Description + "\n"
+			}
+			if config.PluginDefinition.Documentation != "" {
+				message += "\n" + config.PluginDefinition.Documentation + "\n"
+			}
+			configOptions := ""
+			for _, v := range config.PluginConfig {
+				valueType := v.Type
+				if v.Type == "" {
+					valueType = "str"
+				}
+				var optional string
+				if v.Optional {
+					optional = " (optional)"
+				}
+				configOptions += fmt.Sprintf("${GREEN}   %v${RESET}:${BOLD_BLUE}%v${RESET}${WHITE}%v${RESET} Default value: ${BLUE}%v${RESET}\n",
+					strings.ToLower(v.ConfigKey),
+					valueType,
+					optional,
+					v.DefaultValue)
+			}
+			if configOptions != "" {
+				message += "\n${YELLOW}This plugin has the following options:${RESET}\n" + configOptions
+			}
+
+			p := asp.NewParser(state)
+			buildFuncMap := map[string]*asp.Statement{}
```

Comment:
I wonder if we can move some of these things out into their own functions to clean this up a bit e.g. `buildFuncMap = readBuildFunction(state.Config.PluginDefinition.BuildDefsDir)`